runtime/src/accounts_index.rs

```diff
@@ -581,6 +685,53 @@ impl<T: 'static + Clone> AccountsIndex<T> {
         max_root
     }
 
+    pub fn update_secondary_indexes(
+        &self,
+        pubkey: &Pubkey,
+        account_owner: &Pubkey,
+        account_data: &[u8],
+    ) {
+        if *account_owner == inline_spl_token_v2_0::id()
+            && account_data.len() == inline_spl_token_v2_0::state::Account::get_packed_len()
+        {
+            let mint_key = Pubkey::new(&account_data[..PUBKEY_BYTES]);
+            {
+                let mint_account_keys = self.mint_index.index.get(&mint_key).unwrap_or_else(|| {
+                    let res = self
+                        .mint_index
+                        .index
+                        .entry(mint_key)
+                        .or_insert(DashSet::new())
+                        .downgrade();
+                    datapoint_info!(
+                        "mint_index_len",
+                        ("mint_count", self.mint_index.index.len() as i64, i64),
+                    );
+                    res
+                });
+                mint_account_keys.insert(*pubkey);
+            }
+
+            {
+                let account_to_mint = self
+                    .mint_index
+                    .accounts_to_mints
+                    .get(&pubkey)
+                    .unwrap_or_else(|| {
+                        self.mint_index
+                            .accounts_to_mints
+                            .entry(*pubkey)
+                            .or_insert(DashSet::new())
+                            .downgrade()
+                    });
+
+                if account_to_mint.insert(mint_key) && account_to_mint.len() > 1 {
+                    datapoint_info!("key_multiple_mints", ("pubkey", pubkey.to_string(), String),);
+                }
+            }
+        }
```

Comment:
maybe, purge entries if `account.lamports == 0` to avoid cache poisoning?

Reply:
I've updated in the latest commit https://github.com/solana-labs/solana/pull/14212/commits/f4f906a9531d1ff9565996cd8848c712a58d9f68 to track slots in the index, so `purge_keys()` and `purge_keys_exact()` should take care of zero-lamport accounts similar to the regular index path.